static/js/components/Periodogram.jsx

```diff
@@ -0,0 +1,573 @@
+    ];
+
+    g.updateOptions({
+      underlayCallback(canvas) {
+        for (let i = 0; i < lines.length; i += 1) {
+          const loc = Math.log10(x * lines[i][0]);
+          canvas.beginPath();
+          canvas.moveTo(g.toDomXCoord(loc), g.toDomYCoord(0));
+          canvas.lineTo(g.toDomXCoord(loc), g.toDomYCoord(10));
+          [canvas.strokeStyle, canvas.lineWidth] = [lines[i][1], lines[i][2]];
+          canvas.stroke();
+        }
+      },
+    });
+  }
+
+  function ma(minDate, maxDate) {
+    setMjdmin(Math.min(minDate, maxDate));
+    setMjdmax(Math.max(minDate, maxDate));
+  }
+
+  function plotdata(xx, yy, me) {
+    const dat = transpose([xx, yy])
+      .map((x, i) => [x[0], [x[1], me[i]]])
+      .sort((a, b) => b[0] - a[0]);
+    // eslint-disable-next-line no-new
+    new Dygraph(dataplotRef.current, dat, {
+      drawPoints: true,
+      strokeWidth: 0,
+      labels: ["time", "mag"],
+      errorBars: true,
+      valueRange: [Math.max(...yy) + 0.1, Math.min(...yy) - 0.1],
+      dateWindow: [Math.min(...xx) - 6, Math.max(...xx) + 6],
+      zoomCallback: ma,
+    });
+  }
+
+  function plotphased(xx, yy, p, title) {
+    // Create graph with native array as data source
+    const pp = [...xx.map((x) => x % p), ...xx.map((x) => (x % p) + p)];
+    // eslint-disable-next-line no-new
+    new Dygraph(phaseplotRef.current, transpose([pp, [...yy, ...yy]]), {
+      drawPoints: true,
+      strokeWidth: 0,
+      labels: ["phase", "mag"],
+      valueRange: [Math.max(...yy) + 0.1, Math.min(...yy) - 0.1],
+      dateWindow: [0, 2 * p],
+      title,
+    });
+  }
+
+  function plotGLS(xx, yy, kwargs) {
+    // Create graph with native array as data source
+
+    // order by period ([day])
+    const periods_all = xx.map((x) => Math.log10(1.0 / x)).reverse();
+    const power_all = yy.reverse();
+
+    // select only the range we want (< periodmax)
+    const goodi = periods_all
+      .map((e, i) => (e < kwargs.periodmax ? i : undefined))
+      .filter((x) => x);
+    const periods = goodi.map((i) => periods_all[i]);
+    const power = goodi.map((i) => power_all[i]);
+
+    const g = new Dygraph(glsplotRef.current, transpose([periods, power]), {
+      clickCallback(e) {
+        const x = e.offsetX;
+        const y = e.offsetY;
+        const dataXY = g.toDataCoords(x, y);
+        setBestp(10 ** dataXY[0]);
+      },
+      axes: {
+        x: { logscale: false },
+        y: {},
+      },
+      strokeWidth: 2,
+      xlabel: "log Period [day]",
+      ylabel: "Power",
+      labels: ["log Period [day]", "Power"],
+    });
+
+    plotline(g, kwargs.periodbest);
+  }
+
+  useEffect(() => {
+    if (!photometry) {
+      dispatch(photometryActions.fetchSourcePhotometry(id));
+    }
+    if (photometry) {
+      setInstruments([...new Set(photometry.map((x) => x.instrument_name))]);
+      setFilters([...new Set(photometry.map((x) => x.filter))]);
+    }
+
+    if (photometry && !run) {
```

Comment:
move new `if (!run) {}` block to inside upper block (where `photometry` evaluates to `true`)